File: app/src/main/java/mx/dev/franco/automusictagfixer/di/MediaPlayerModule.java
package mx.dev.franco.automusictagfixer.di;

import android.app.Application;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import mx.dev.franco.automusictagfixer.audioplayer.SimpleMediaPlayer;

@Module
public class MediaPlayerModule {
    @Singleton
    @Provides
    SimpleMediaPlayer providesMediaPlayer(Application application) {
        return SimpleMediaPlayer.getInstance(application);
    }
}

File: app/src/main/java/mx/dev/franco/automusictagfixer/di/MediaPlayerModule.kt
package mx.dev.franco.automusictagfixer.di

import android.content.Context
import dagger.Module
import dagger.Provides
import mx.dev.franco.automusictagfixer.audioplayer.SimpleMediaPlayer
import javax.inject.Singleton

@Module
class MediaPlayerModule {

    @Singleton
    @Provides
    fun providesMediaPlayer(application: Context) = SimpleMediaPlayer.getInstance(application)

}